libc-top-half/musl/src/thread/pthread_create.c

```diff
@@ -60,6 +60,18 @@ void __tl_sync(pthread_t td)
 	if (tl_lock_waiters) __wake(&__thread_list_lock, 1, 0);
 }
 
+#ifndef __wasilibc_unmodified_upstream
+/* this queue has at most one entry */
```

Comment:
Give that, I'm not sure `queue` is the right name for this.

I'm not sure I have any better suggestions though.  Maybe `deferred_deallocation` and `process_deferred_deallocation`?

Reply:
renamed